src/main.rs

```diff
@@ -267,7 +267,32 @@ impl EventObserver for KeyLogger {
 	}
 }
 
+fn check_uinput_loaded() -> bool {
+	let mut kernel_version = std::fs::read_to_string("/proc/version").unwrap();
+	kernel_version = kernel_version.to_string().split(' ').nth(2).unwrap().to_string();
+
+	let built_in_modules = std::fs::read_to_string(String::from("/lib/modules/") + &kernel_version + &String::from("/modules.builtin")).unwrap();
+	for line in built_in_modules.lines() {
+		if line.eq("kernel/drivers/input/misc/uinput.ko") {
+			info!("'uinput' is built into running kernel version: {}", kernel_version);
+			return true;
+		}
+	}
+
+	let modules = std::fs::read_to_string("/proc/modules").unwrap();
+	for line in modules.lines() {
+		if line.starts_with("uinput ") {
+			info!("'uinput' module is loaded");
+			return true;
+		}
+	}
+	return false;
+}
+
```

Comment:
Thanks for the PR

How confident are you about this being cross-platform? Will it work across most popular linux desktop distros/kernel versions?

Reply:
I am confident only that it compiles cleanly on Arch. Your question is very fair. I have not tested it on other Linux distros/kernels.

Reply:
compilation is of course not the concern. Main concern is runtime failure (based on distro, kernel version etc..). I think this should be handled in a wrapper script.

Reply:
Do you mean that you want a wrapper script in the `kbct` codebase or that a wrapper script should be added by distros when packaging `kbct`?

Reply:
By distros, for example for systemd you can add pre-init script that will load kernel module.